docs/reference/mapping.asciidoc

```diff
@@ -93,6 +93,13 @@ causing a mapping explosion:
 `index.mapping.nested_objects.limit`::
     The maximum number of `nested` JSON objects within a single document across
     all nested types, defaults to 10000.
++
+[IMPORTANT]
+====
+If you increase this setting, we recommend you also increase the
+<<search-settings,`indices.query.bool.max_clause_count`>> setting, which
+limits the maximum number of <<query-dsl-bool-query,boolean clauses>> in a query.
+====
```

Comment:
This limit is for the number of nested docs that are allowed on a single nested field. So by default we limit the number of nested documents in the array to 10,000 but that is not related to the boolean max clause. 
I'd put the same sentence but in the description of `index.mapping.total_fields.limit` with a warning that too many fields in mapping and queries should be avoided.

Reply:
Thanks @jimczi. I've relocated the note to `index.mapping.total_fields.limit` with a2eff0807be.